src/queries/user/badges.ts

```diff
@@ -9,7 +9,14 @@ const resolver: GQLUserInfoResolvers['badges'] = async (
     return []
   }
 
-  return atomService.findMany({ table: 'user_badge', where: { userId: id } })
+  const badges = await atomService.findMany({
+    table: 'user_badge',
+    where: { userId: id },
+  })
+
+  return badges.map(({ type, extra }) => ({
+    type: `${type}${extra?.level || ''}` as GQLBadgeType,
```

Comment:
`extra` is arbitrary, how can it strong typed here?

Reply:
we can do application level value clamp here; a typical use in DocumentDB